src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -229,16 +224,15 @@ public final void close() throws SQLException
          try {
             if (isCommitStateDirty && !isAutoCommit && !isReadOnly) {
                delegate.rollback();
-               lastAccess = clockSource.currentTime();
                LOGGER.debug("{} - Executed rollback on connection {} due to dirty commit state on close().", poolEntry.getPoolName(), delegate);
             }
 
             if (dirtyBits != 0) {
                poolEntry.resetConnectionState(this, dirtyBits);
-               lastAccess = clockSource.currentTime();
             }
 
             delegate.clearWarnings();
+            lastAccess = clockSource.currentTime();
          }
```

Comment:
setting lastAccess here.. only after 'successful' any sql calls before and lastly clearWarnings() ...closest before recycling